xonsh/vox.py

```diff
@@ -0,0 +1,197 @@
+from os.path import join, basename, exists
+import os
+from shutil import rmtree
+import venv
+
+import builtins
+
+
+class Vox:
+    """Vox is a virtual environment manager for xonsh."""
+
+    def __init__(self):
+        """Ensure that $VIRTUALENV_HOME is defined and declare the available vox commands"""
+
+        if not builtins.__xonsh_env__.get('VIRTUALENV_HOME'):
+            if os.name == 'nt':
+                home_path = builtins.__xonsh_env__['USERPROFILE']
+
+            elif os.name == 'posix':
+                home_path = builtins.__xonsh_env__['HOME']
+
+            else:
+                print('This OS is not supported.')
+                return None
+
+            builtins.__xonsh_env__['VIRTUALENV_HOME'] = join(home_path, '.virtualenvs')
+
+        self.commands = {
+            ('new',): self.new,
+            ('activate', 'workon', 'enter'): self.activate,
+            ('deactivate', 'exit'): self.deactivate,
+            ('list', 'ls'): self.list,
+            ('remove', 'rm', 'delete', 'del'): self.remove,
+            ('help', '-h', '--help'): self.help
+        }
+
+    def __call__(self, args, stdin=None):
+        """Call the right handler method for a given command."""
+
+        if not args:
+            self.help()
+            return None
+
+        command_name, params = args[0], args[1:]
+
+        try:
+            command = [
+                self.commands[aliases] for aliases in self.commands
+                if command_name in aliases
+            ][0]
+
+            command(*params)
+
+        except IndexError:
+            print('Command "%s" doesn\'t exist.\n' % command_name)
+            self.print_commands()
+
+    @staticmethod
+    def new(name):
+        """Create a virtual environment in $VIRTUALENV_HOME with ``python3 -m venv``.
+
+        :param name: virtual environment name
+        """
+
+        env_path = join(builtins.__xonsh_env__['VIRTUALENV_HOME'], name)
+
+        print('Creating environment...')
+
+        vbuiltins.__xonsh_env__.create(env_path, with_pip=True)
```

Comment:
Thanks a lot for the correction! It's supposed to be `venv`. When I autoreplaced ENV with builtins.__xonsh_env__, I totally overlooked this bit :-(

Fixed in https://github.com/moigagoo/xonsh/commit/b92969ec4b4aa134d7b91ee5f7b4e98d8afe80c2